x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/types.ts

```diff
@@ -45,5 +45,5 @@ export interface CurationResult {
   // TODO: Consider updating our internal API to return more standard Result data in the future
   id: string;
   _meta?: ResultMeta;
-  [key: string]: string | string[] | ResultMeta | undefined;
+  [key: string]: string | string[] | ResultMeta | object | undefined;
```

Comment:
I struggled with that for a while, but I don't have enough knowledge of TypeScript to make it work. I need an `object` to call `object.keys` or `object.entries`.

Reply:
If you replace the `object` with `Record<string, unknown>` it should just work. It's just explicit about it being a string-indexed object with unknown values.

Reply:
I have checked it locally, so we have an issue with the types created before. 
I would ask @sphilipse 's feedback here as I cannot think a simpler approach other than overhauling the `Result` and `CurationResult` types.

What I tried to do is to change the input/output type of the utility functions to conform the `Record<string, unknown>`, but when we try to use the same function for both types, there will be an issue one place or another because on `Result` type we have assigned `[key: string]: object` and we are trying to avoid doing it on `CurationResult` type. Probably there is a better, simpler approach but I cannot think any at the moment.

Reply:
There's no great solution here because of the way dictionary object types work, which is: they don't work when some of the keys on the object are *not* dictionary entries.

If I look at the code now it always returns CurationResult so we don't really need type guards/narrowing to fix the typing for Result vs CurationResult. It just gives back CurationResult.

There's two ways forward imo:
1. Turn Result and/or CurationResult into something of the shape: 
```
{
  id: string,
  _meta: MetaStuff,
  contents: Record<string, unknown>,
}
```
so that we can use the contents property for the dictionary stuff. And then use generics to fix all the typing everywhere. 
2. Say "screw it, it works, we don't NEED it to be perfect"

At this point I'd lean towards 2